.github/workflows/submit-contract-sizes.yml

```diff
@@ -0,0 +1,46 @@
+# The workflow has write access, so it needs to be isolated for security reasons from pull request-based workflows,
+# which may be triggered from forked repositories.
+on:
+  workflow_run:
+    workflows:
+      - continuous-integration
+    types:
+      - completed
+
+jobs:
+  submit-contract-sizes:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+        github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Download Contract Sizes
+        uses: ./.github/download-artifact
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ARTIFACT_NAME: contract-sizes-data
+
+
+      - name: Submit Comment
+        env:
+          GITHUB_PR_TOKEN:         ${{ secrets.github_token }}
+          GITHUB_PR_WORKFLOW_ID:   ${{ github.event.workflow_run.id }}
+        run: |
+          # context.out is considered as an untrusted file
+          PR_NUMBER=$(grep -oE 'PR_NUMBER="[^"]+"' context.out | awk -F '"' '{print $2}')
+          CARGO_CONTRACT_VERSION=$(grep -oE 'CARGO_CONTRACT_VERSION="[^"]+"' context.out | awk -F '"' '{print $2}')
+          PR_COMMENTS_URL="${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/issues/${PR_NUMBER}/comments"
+          WORKFLOW_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_PR_WORKFLOW_ID}"
+
+          # Submit the comparison table as a comment to the PR
+          echo "Submitting contract sizes diff to ${PR_COMMENTS_URL}"
+          GITHUB_PR_TOKEN=${GITHUB_PR_TOKEN} scripts/contract_sizes_submit.sh ${PR_COMMENTS_URL} ${WORKFLOW_URL} ${CARGO_CONTRACT_VERSION} < ./contract_sizes_diff.md
```

Comment:
So `contract_sizes_diff.md` comes from the downloaded `contract-sizes-data`? It's not 100% clear and relying on the filename is brittle. Are we able to paramerize this somehow?

Reply:
contract_sizes_diff.md along with context.out is part of contract-sizes-data artifact.
I can add comment or put it to env - but env is not shared between workflows.

Reply:
Would it not be possible to structure this `contract-sizes` as a single job, and then `submit` as a step, and then use [`$GITHUB_OUTPUT`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) to send values between the individual steps, in this case it would contain the contents of `contract_sizes_diff.md`

Reply:
It is possible, I could user here pull_request_target trigger but it introduces security risk.
"GitHub’s standard pull_request workflow trigger by default prevents write permissions and secrets access to the target repository. However, in some scenarios such access is needed to properly process the PR. To this end the pull_request_target workflow trigger was introduced." 
More informations: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/